fea-rs/src/compile/error.rs

```diff
@@ -39,3 +41,64 @@ pub enum GlyphOrderError {
     #[error("The first glyph must be '.notdef'")]
     MissingNotDef,
 }
+
+/// An error reported by the compiler
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum CompilerError {
```

Comment:
It's slightly tiresome that CompilerError is not Send. The result is that if I try to make the naive error to wrap this:

```rust
    #[error("Fea compilation failure")]
    FeaCompileError(#[from] CompilerError),
```

I get yelled at:

```shell
error[E0277]: `Rc<(dyn StdError + 'static)>` cannot be sent between threads safely
   --> fontc/src/main.rs:409:33
    |
409 |                       scope.spawn(move |_| {
    |  ___________________________-----_^
    | |                           |
    | |                           required by a bound introduced by this call
410 | |                         let result = work.exec(work_context);
411 | |                         if let Err(e) = send.send((id.clone(), result)) {
412 | |                             error!("Unable to write {:?} to completion channel: {}", id, e);
413 | |                         }
414 | |                     })
    | |_____________________^ `Rc<(dyn StdError + 'static)>` cannot be sent between threads safely
    |
    = help: within `(AnyWorkId, Result<(), AnyWorkError>)`, the trait `Send` is not implemented for `Rc<(dyn StdError + 'static)>`
    = note: required because it appears within the type `fea_rs::parse::source::SourceLoadError`
    = note: required because it appears within the type `fea_rs::compile::error::CompilerError`
    = note: required because it appears within the type `fontbe::error::Error`
    = note: required because it appears within the type `AnyWorkError`
    = note: required because it appears within the type `Result<(), AnyWorkError>`
    = note: required because it appears within the type `(AnyWorkId, Result<(), AnyWorkError>)`
    = note: required for `crossbeam_channel::Sender<(AnyWorkId, Result<(), AnyWorkError>)>` to implement `Send`
note: required because it's used within this closure
   --> fontc/src/main.rs:409:33
````

(the actual output is rather longer but it's just repetitively sharing that it would quite like it if we could be Send)